core/src/main/java/org/elasticsearch/index/query/FilteredQueryParser.java

```diff
@@ -43,13 +40,13 @@ public FilteredQueryParser() {
         return new String[]{FilteredQueryBuilder.NAME};
     }
 
+    @SuppressWarnings("rawtypes")
     @Override
-    public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+    public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOException, QueryParsingException {
         XContentParser parser = parseContext.parser();
 
-        Query query = Queries.newMatchAllQuery();
-        Query filter = null;
-        boolean filterFound = false;
+        QueryBuilder<MatchAllQueryBuilder> query = new MatchAllQueryBuilder();
```

Comment:
the use of generics here is fishy. You wanted to assign a default value to the query I think? But then the query builder that gets parsed later on does not necessarilyhave MatchAllQueryBuilder type?